protected/components/DownloadService.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+/**
+ * Service to provide tokens for password reset functionality
+ */
+class DownloadService extends yii\base\Component
+{
+    /**
+     * Initializes application component.
+     * 
+     * This method overrides the parent implementation by setting default cache
+     * key prefix.
+     */
+    public function init()
+    {
+        parent::init();
+    }
+
+    /**
+     * Downloads and returns contents of a remote file
+     */
+    public static function downloadFile(string $url)
+    {
+        $curl = new Curl\Curl();
```

Comment:
Re the HTTP client, 

Are there some reasons you didn't use [Guzzlehttp](https://packagist.org/packages/guzzlehttp/guzzle) which is used in a lot of places in the codebase (especially services classes like yours) and is already listed in the ``composer.json`` and ``composer.lock`` file?

Furthermore, Guzzlehttp is pretty much the industry standard for modern  PHP as far as http clients are concerned (so if you scan our Composer dependencies, you will see that's the http client used the most and by far), and it supports all of the relevant PSRs which Curl don't.

Also, you wouldn't need to meddle with additional PHP extensions in the Dockerfile by doing away with Curl.

Reply:
I didn't know Guzzlehttp is the industry standard. Guzzlehttp is now used in the code.